test/fixtures/framework/features/steps/steps.rb

```diff
@@ -1,3 +1,14 @@
+When('I fail on the first attempt') do
+  if $first_attempt
+    $first_attempt = false
+    fail 'Failing on the first attempt'
+  end
+end
+
+When('I let the scenario retry') do
```

Comment:
Should this be moved into the main steps?

Reply:
It was intentional to have it just in the localised test steps, for a couple of reasons:

"I let the scenario retry" isn't needed in general - if a blanket retry is allowed then we already have `@retry`.

The way I saw the feature working was for `Maze.dynamic_retry` to be the API for test suites to work with.  Dynamic retries should only be allowed in specific circumstances, and I couldn't see a way to encapsulate that in a generic step.  For example, the reason I'm adding this feature is to allow an exception with a specific message to occur, but not any other.

Reply:
Okay, that makes sense